controllers/topics/index.js

```diff
@@ -337,12 +337,13 @@ const followTopicV2 = async (req, res) => {
 
 const _topicAutoMessage = async (user_id, topic_id, detailTokenUser) => {
   try {
-    const communityMessageFormat = await CommunityMessageFormat.findOne({
+    const communityMessageFormats = await CommunityMessageFormat.findAll({
       where: {topic_id, status: '1'}
     });
-
-    if (communityMessageFormat && !detailTokenUser.is_anonymous) {
-      followTopicServiceQueue(user_id, topic_id);
+    if (communityMessageFormats && !detailTokenUser.is_anonymous) {
+      for (const format of communityMessageFormats) {
+        followTopicServiceQueue(user_id, topic_id, format.delay_time);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `followTopicServiceQueue` function is being called inside a loop without `await`. This could lead to unexpected behavior if the function is asynchronous, as it would not wait for the previous call to finish before starting the next one. If `followTopicServiceQueue` is an async function, consider using `Promise.all` to handle all promises at once or use `for...of` with `await` to handle them sequentially.

```diff
- for (const format of communityMessageFormats) {
-   followTopicServiceQueue(user_id, topic_id, format.delay_time);
- }
+ const promises = communityMessageFormats.map(format => 
+   followTopicServiceQueue(user_id, topic_id, format.delay_time)
+ );
+ await Promise.all(promises);
```

Also, error handling seems to be missing for the `followTopicServiceQueue` function. If this function throws an error, it will not be caught and handled properly. Consider adding try-catch blocks around it or handle the promise rejection.

```diff
+ const promises = communityMessageFormats.map(format => {
+   try {
+     return followTopicServiceQueue(user_id, topic_id, format.delay_time);
+   } catch (error) {
+     console.error(`Error in followTopicServiceQueue: ${error}`);
+   }
+ });
+ await Promise.all(promises);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->